### PR TITLE
Ehub 129

### PR DESCRIPTION
**Pull request by vijayyadav10**

### CODE REVIEW

**Review by ShreyasBhargavaNV — CHANGES_REQUESTED** [review]

File: application/ui/widgets/eh-widgets-dir/eh-widgets/src/page/catalog/catalog-tile/modal-delete-bundle-group/ModalDeleteBundleGroup.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                open={open}
+                onRequestClose={onRequestClose}
+                onRequestSubmit={onRequestDelete}
+            >
+                <div className="Modal-delete-bundle-group">
+                    <div className="Modal-delete-bundle-group-wrapper">
+                        <TrashCan32 />
+                    </div>
+                    <div>
+                        Are you sure you want to delete this bundle?
```
[the remark this answers is not in the document]
Comment (vijayyadav10) [reply application/ui/widgets/eh-widgets-dir/eh-widgets/src/page/catalog/catalog-tile/modal-delete-bundle-group/ModalDeleteBundleGroup.js]: done as suggested.

File: application/ui/widgets/eh-widgets-dir/eh-widgets/src/page/catalog/catalog-tile/modal-update-bundle-group/ModalUpdateBundleGroup.js [hunk 2]
```diff
@@ -109,15 +113,24 @@ export const ModalUpdateBundleGroup = ({
...
       await bundleGroupSchema
-        .validate(bundleGroup, { abortEarly: false })
-        .catch((err) => {
-          validationError = fillErrors(err)
-        })
+      .validate(bundleGroup, { abortEarly: false })
+      .catch((err) => {
+        validationError = fillErrors(err)
+      })
+      if ((bundleGroup.status === BUNDLE_STATUS.NOT_PUBLISHED || bundleGroup.status === BUNDLE_STATUS.DELETE_REQ) &&
```
[the remark this answers is not in the document]
Comment (vijayyadav10) [reply application/ui/widgets/eh-widgets-dir/eh-widgets/src/page/catalog/catalog-tile/modal-update-bundle-group/ModalUpdateBundleGroup.js]: done as suggested(added null checks)

File: application/ui/widgets/eh-widgets-dir/eh-widgets/src/page/catalog/catalog-tile/overflow-menu/CatalogTileOverflowMenu.js [hunk 3]
```diff
@@ -1,17 +1,31 @@
...
 
-const CatalogTileOverflowMenu = ({bundleGroupId, onAfterSubmit}) => {
+const CatalogTileOverflowMenu = ({bundleGroupId, bundleStatus, bundleName,onAfterSubmit}) => {
 
     const [openModal, setOpenModal] = useState(false)
+    const [deleteModal, setDeleteModal] = useState(false)
+    const higherRole = getHigherRole()
+
+    const isShowDelete = (higherRole === MANAGER || higherRole === ADMIN) ? true : false;
+    const isDeletableBundle = bundleStatus === 'DELETE_REQ' ? true : false
```
[the remark this answers is not in the document]
Comment (vijayyadav10) [reply application/ui/widgets/eh-widgets-dir/eh-widgets/src/page/catalog/catalog-tile/overflow-menu/CatalogTileOverflowMenu.js]: added to constant file

File: application/ui/widgets/eh-widgets-dir/eh-widgets/src/integration/Integration.js [hunk 4]
```diff
@@ -300,6 +300,18 @@ export const editBundleGroup = async (bundleGroupData, id) => {
   return checkForErrorsAndSendResponse(data, isError, "editedBundleGroup")
```
[the remark this answers is not in the document]
Comment (vijayyadav10) [reply application/ui/widgets/eh-widgets-dir/eh-widgets/src/integration/Integration.js]: added to constant file

File: application/ui/widgets/eh-widgets-dir/eh-widgets/src/integration/Integration.js [hunk 5]
```diff
@@ -297,7 +298,19 @@ export const editBundleGroup = async (bundleGroupData, id) => {
...
+export const deleteBundle = async (id, bundleName) => {
+  const { data, isError } = await deleteData(urlBundleGroups, id)
+
+  eventHandler(
+    isError,
+    `Impossible to delete bundle. ${data ? data.message : ""}`,
+    `Bundle ${bundleName ? bundleName : ""} deleted`
+  )
+
+  return checkForErrorsAndSendResponse(data, isError, "deletedBundle")
```
Comment (ShreyasBhargavaNV) [inline application/ui/widgets/eh-widgets-dir/eh-widgets/src/integration/Integration.js:313]: deletedBundle this should come from constant file

File: application/ui/widgets/eh-widgets-dir/eh-widgets/src/page/catalog/catalog-tile/modal-delete-bundle-group/ModalDeleteBundleGroup.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        onAfterSubmit()
+    }
+
+    return (
+        <>
+            <Modal
+                className="Modal-edit-bundle-group"
+                modalLabel={BUTTON_LABELS.DELETE}
+                primaryButtonText={BUTTON_LABELS.DELETE}
+                secondaryButtonText="Cancel"
```
Comment (ShreyasBhargavaNV) [inline application/ui/widgets/eh-widgets-dir/eh-widgets/src/page/catalog/catalog-tile/modal-delete-bundle-group/ModalDeleteBundleGroup.js:26]: "Cancle" should come from Constant

File: application/ui/widgets/eh-widgets-dir/eh-widgets/src/components/forms/BundleGroupForm/update-boundle-group/bundles-of-bundle-group/BundlesOfBundleGroup.js [hunk 7]
```diff
@@ -128,7 +128,7 @@ const BundlesOfBundleGroup = ({
 
     if (minOneBundleError === MIN_ONE_BUNDLE_ERROR &&
         Object.keys(validationResult).length === 0 &&
-        initialBundleList.length < 1 && bundleStatus !== BUNDLE_STATUS.NOT_PUBLISHED) {
+        initialBundleList.length < 1 && (bundleStatus === BUNDLE_STATUS.PUBLISHED || bundleStatus === BUNDLE_STATUS.PUBLISH_REQ)) {
         bundleUrlErrorResult = MIN_ONE_BUNDLE_ERROR;
     } else {
         bundleUrlErrorResult = validationResult["gitRepo"] &&
```
Comment (ShreyasBhargavaNV) [inline application/ui/widgets/eh-widgets-dir/eh-widgets/src/components/forms/BundleGroupForm/update-boundle-group/bundles-of-bundle-group/BundlesOfBundleGroup.js:134]: gitRepo should come from constant file

**Review by ShreyasBhargavaNV — APPROVED** [review]

**Pull request merged**